crates/extensions/c8y_config_manager/src/config.rs

```diff
@@ -51,6 +53,8 @@ impl ConfigManagerConfig {
 
         let plugin_config = PluginConfig::new(&plugin_config_path);
 
+        let file_transfer_dir = data_dir.join(DEFAULT_FILE_TRANSFER_DIR_NAME);
```

Comment:
That constant `DEFAULT_FILE_TRANSFER_DIR_NAME` is **not** defined here but rather referred from the shared `tedge_config_defaults` module. It can't be defined as a single constant(`/var/tedge/file-transfer`) in the `http_rest` module as the location is relative to the `data.dir` location which needs to be read from the tedge config. That's why I'm passing the `data_dir` as an argument everywhere and forced to recreate this `file_transfer_dir` path from the `data_dir` in all the plugins.

Reply:
Yeah, that's an option.

Reply:
> Yeah, that's an option. I can add another read-only `FileTransferPathSetting` similar to the `DeviceIdSetting`.

Maybe as a follow-up PR, then. We have first to define the future of the `tedge_config` crate.